Week11/app/src/main/java/com/example/marvelcharacters/view/characterdetail/CharacterDetailActivity.kt

```diff
@@ -0,0 +1,60 @@
+package com.example.marvelcharacters.view.characterdetail
+
+import android.content.Context
+import android.content.Intent
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import com.bumptech.glide.Glide
+import com.example.marvelcharacters.R
+import com.example.marvelcharacters.model.entities.Character
+import com.example.marvelcharacters.model.entities.Thumbnail
+import kotlinx.android.synthetic.main.activity_character_detail.*
+import kotlinx.android.synthetic.main.character_view_holder.view.*
+
+fun Context.characterDetailActivityIntent(
+    characterName: String,
+    characterDescription: String,
+    characterThumbnail: Thumbnail
+): Intent {
+    return Intent(this, CharacterDetailActivity::class.java).apply {
+        putExtra(INTENT_CHARACTER_NAME, characterName)
+        putExtra(INTENT_CHARACTER_DESCRIPTION, characterDescription)
+        putExtra(INTENT_CHARACTER_THUMBNAIL, characterThumbnail)
+    }
+}
+
+private const val INTENT_CHARACTER_NAME = "character_name"
+private const val INTENT_CHARACTER_DESCRIPTION = "character_description"
+private const val INTENT_CHARACTER_THUMBNAIL = "character_thumbnail"
+
+class CharacterDetailActivity : AppCompatActivity() {
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_character_detail)
+
+        val characterName = intent.getStringExtra(INTENT_CHARACTER_NAME)
+        val characterDescription = intent.getStringExtra(INTENT_CHARACTER_DESCRIPTION)
+        val characterThumbnail = intent.getParcelableExtra<Thumbnail>(INTENT_CHARACTER_THUMBNAIL)
+
+        requireNotNull(characterName) { "no characterName provided in Intent extras" }
```

Comment:
You don't really want to crash the app if there's no character name. Just provide some kind of a placeholder or info to the user.